src/controllers/userAuth.ts

```diff
@@ -197,6 +197,50 @@ const userAuthController = {
                 }
             });
         }
+    },
+
+    resetPassword: async (req: Request, res: Response) => {
+        const errors = validationResult(req);
+
+        if (!errors.isEmpty()) {
+            res.status(statusCodes.MISSING_PARAMS).json(
+                {
+                    status: 422,
+                    message: `Error: there are missing parameters.`
+                }
+            );
+        } else {
+            const oldPassword = req.body.oldPassword;
+            const newPassword = req.body.newPassword;
+            const jwtToken = req.body.jwtToken;
+
+            jwt.verify(jwtToken, process.env.JWT_SECRET, async (err, decoded) => {
+                if (err) {
+                    res.status(statusCodes.BAD_REQUEST).json(
+                        {
+                            status: 422,
+                            message: err
+                        }
+                    );
+                } else {
+                    let userData : IUserModel;
+                    const userId = decoded._id;
+                    try {
+                        userData = await userDBInteractions.findWithPassword(userId);
+                        const result = await compareHash(oldPassword, userData.password);;
+                        if (!result) {
+                            throw new Error('Wrong old password');
```

Comment:
this wont print in .json(error)
either `throw { "error" : "Wrong old password" `} or  `                    res.status(statusCodes.BAD_REQUEST).json(
                        {
                            status: statusCodes.UNAUTHORIZED,
                            message: 'Wrong old password'
                        }
                    );`

Reply:
Sounds good will change that